src/unix/pipe.c

```diff
@@ -302,3 +302,68 @@ uv_handle_type uv_pipe_pending_type(uv_pipe_t* handle) {
   else
     return uv__handle_type(handle->accepted_fd);
 }
+
+
+int uv_pipe_chmod(uv_pipe_t* handle, int mode) {
+  #define UV__WRITABLE_DEF (S_IWUSR | S_IWGRP | S_IWOTH)
+  #define UV__READABLE_DEF (S_IRUSR | S_IRGRP | S_IROTH)
+  struct stat pipe_stat;
+  char* name_buffer;
+  size_t name_len;
+  int r;
+
+  if (!handle || uv__stream_fd(handle) == -1)
+    return UV_EBADF;
+
+  if (mode != UV_READABLE &&
+      mode != UV_WRITABLE &&
+      mode != (UV_WRITABLE | UV_READABLE))
+    return UV_EINVAL;
+
+  if (fstat(uv__stream_fd(handle), &pipe_stat) == -1)
+    return -errno;
+
+  /* Exit early if pipe already has desired mode. */
+  if (mode & UV_READABLE && 
+      (pipe_stat.st_mode & UV__READABLE_DEF) == UV__READABLE_DEF)
+    return 0;
+  if (mode & UV_WRITABLE &&
+      (pipe_stat.st_mode & UV__WRITABLE_DEF) == UV__WRITABLE_DEF)
+    return 0;
```

Comment:
Is this correct? What if `mode` is `UV_READABLE | UV WRITABLE` and `pipe_stat.st_mode` already contains the readable flags but not the writable?